src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -234,25 +235,27 @@ internal void ApplyHttpsDefaults(HttpsConnectionAdapterOptions httpsOptions)
     /// </summary>
     public void Load()
     {
-        if (_loaded)
+        if (_reloadToken is null || _reloadToken.HasChanged)
```

Comment:
Should this honor `!ReloadOnChange` and not reload?

Reply:
I understood that property to control _implicit_ reloads.  Surely, if you call `Load`, you actually want it to reflect the latest values.

Reply:
I'm a little hesitant to start using the token in any scenario where we didn't at all before because you never know what `GetReloadToken()` implementations might throw or otherwise misbehave. If it's our view that an extra call to `Load()` is an explicit signal, maybe we should always reread configuration in `Load()` without checking the token. 

I think the safest solution is to keep the `_loaded` bool and continue to no-op after the first call to the public `Load()` method.  `UseHttps()` and `KestrelServerImpl` can call internal methods that check the `_reloadToken` only in the `ReloadOnChange` case. Having `UseHttps()` call the `EndpointsToAdd` callbacks and potentially raise exceptions from them would be weird anyway.

Reply:
I'm going to approve this because it's not a huge deal, but I'd prefer we not call `GetReloadToken()` when we do not need to.

I think the public `Load()` method should continue no-op after the first call for simplicity's sake. I haven't heard anyone ask for it not to. And if we do that, there is no reason to capture the token in the `!ReloadOnChange` case. After we fix `UseHttps()` to eagerly load config by calling some other internal method, there will be no real reason to ever call `Load()` anyway.